v2/yarpcrouter/router.go

```diff
@@ -58,19 +59,78 @@ type MapRouter struct {
 func NewMapRouter(defaultService string) MapRouter {
 	return MapRouter{
 		defaultService:            defaultService,
-		serviceProcedureEncodings: make(map[serviceProcedureEncoding]yarpc.Procedure),
+		serviceProcedureEncodings: make(map[serviceProcedureEncoding]yarpc.TransportProcedure),
 		serviceNames:              map[string]struct{}{defaultService: {}},
 	}
 }
 
+// MapToTransportProcedures converts encoding-level procedures to transport-level procedures
+func MapToTransportProcedures(encodingProcedures []yarpc.EncodingProcedure) []yarpc.TransportProcedure {
+	transportProcedures := []yarpc.TransportProcedure{}
+	for _, p := range encodingProcedures {
+		var transportHandlerSpec yarpc.TransportHandlerSpec
+		switch p.HandlerSpec.Type() {
+		case yarpc.Unary:
```

Comment:
What about streams?

Reply:
I think it would be reasonable to tackle streams in the next pass, if we have a note in CHANGES and don’t break the existing router’s support for streams.

Reply:
I added 

> - [ ] support StreamHandlers in EncodingHandlerSpec

to CHANGES in an older PRe